### PR TITLE
converter

### PR DESCRIPTION
**Pull request by liubinski27**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,16 +1,58 @@
...
+import { ConverterComponent } from './converter/converter.component';
+import { HeaderComponent } from './header/header.component';
+import { FooterComponent } from './footer/footer.component';
+import { CoursesComponent } from './courses/courses.component';
+import { DateComponent } from './date/date.component';
+import { ConverterService } from './converter.service';
+import { UrlService } from './url.service';
+
+const appRoutes: Routes = [
+    { path: '', component: CoursesComponent },
```
Comment (kirill-malyhin) [inline src/app/app.module.ts:19]: should be splitted to separate file

File: src/app/converter.service.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  constructor(
+    private http: HttpClient,
+    private urlService: UrlService
+  ) { }
+
+  getCurrencies(date: string) {
+    const subject = new BehaviorSubject(this.currenciesList);
+    const url = this.urlService.getUrl(date);
+    this.http.get(url).subscribe((response: ICurrencyResponse[]) => {
+      this.currenciesList = response.map(item => ({
```
Comment (kirill-malyhin) [inline src/app/converter.service.ts:25]: move processing into separate function in service

File: src/app/converter/converter.component.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<div class="container-fluid">
+    <app-date (onChanged)="getCurrencies($event)"></app-date>
```
Comment (kirill-malyhin) [inline src/app/converter/converter.component.html:2]: (onChanged) -> (onChange)

File: src/app/converter/converter.component.scss [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    .inputValue {
+        width: 300px;
+        display: flex;
+        justify-content: space-between;
+        align-items: center;
+    }
+
+    .loading {
```
Comment (kirill-malyhin) [inline src/app/converter/converter.component.scss:41]: should be a common class as it is declared also in other place

File: src/app/converter/converter.component.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  }
+
+  deleteInput(selectedCurrencyAbbreviation: string) {
+    const indexOfSelectedCurrency = this.selectedCurrencies.findIndex(item => item.abbreviation === selectedCurrencyAbbreviation);
+    if (indexOfSelectedCurrency >= 0) {
+      this.selectedCurrencies.splice(indexOfSelectedCurrency, 1);
+    }
+    if (!this.selectedCurrencies.length) {
+      this.convertedInputValueInByn = 0;
+      this.savedCurrencyValue = 0;
```
Comment (kirill-malyhin) [inline src/app/converter/converter.component.ts:53]: as we discussed, we don't store value

File: src/app/converter/converter.component.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      this.savedCurrencyIndex = 0;
```
Comment (kirill-malyhin) [inline src/app/converter/converter.component.ts:54]: we store only selected currencies

File: src/app/converter/converter.component.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    } else {
+      const value = this.convertedInputValueInByn * currencyScale / currencyRate;
+      return Number.isInteger(value) ? value : value.toFixed(4);
+    }
+  }
+
+  saveData() {
+    this.selectedCurrenciesAbbreviations = [];
+    this.selectedCurrencies.forEach(item => {
+      const foundCurrencyAbbreviation = this.selectedCurrenciesAbbreviations.find(el => el === item.abbreviation);
```
Comment (kirill-malyhin) [inline src/app/converter/converter.component.ts:82]: foundCurrencyAbbreviation  -> currencyAbbreviation

File: src/app/header/header.component.html [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+                </li>
+                <li class="nav-item">
+                    <div class="changeLang">
+                        <button class="btn btn-default dropdown-toggle" type="button" id="dropdownMenu1"
+                            data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+                            <img src="../../assets/styles/pictures/pngegg.png" alt="language">
+                            <span class="caret"></span>
+                        </button>
+                        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
+                            <li (click)="changeLang('ru')">Русский</li>
```
Comment (kirill-malyhin) [inline src/app/header/header.component.html:25]: 'ru', 'en' move to enum

File: src/app/header/header.component.html [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+                    <a class="nav-link" routerLink="">{{ 'menu.currency.rates' | translate }}</a>
+                </li>
+                <li class="nav-item">
+                    <a class="nav-link" routerLink="/converter">{{ 'menu.currency.converter' | translate }}</a>
+                </li>
+                <li class="nav-item">
+                    <div class="changeLang">
+                        <button class="btn btn-default dropdown-toggle" type="button" id="dropdownMenu1"
+                            data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
+                            <img src="../../assets/styles/pictures/pngegg.png" alt="language">
```
Comment (kirill-malyhin) [inline src/app/header/header.component.html:21]: rename pngegg to something not shitty

File: src/index.html [hunk 10]
```diff
@@ -1,13 +1,16 @@
 <!doctype html>
 <html lang="en">
+
 <head>
   <meta charset="utf-8">
-  <title>CurrencyConverter</title>
+  <title>Currency-Converter</title>
```
Comment (kirill-malyhin) [inline src/index.html:6]: Currency Converter

File: src/app/converter/converter.component.ts (hunk 5 above)
Comment (liubinski27) [reply src/app/converter/converter.component.ts:53]: this is the function for deleting inputs, these variables are not stored

File: src/app/converter/converter.component.ts (hunk 6 above)
Comment (liubinski27) [reply src/app/converter/converter.component.ts:54]: maybe the variables are named incorrectly, as we store only selected currencies

File: src/app/language.service.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+})
+
+export class LanguageService {
+
+  constructor(
+    private translateService: TranslateService
+  ) {
+    translateService.addLangs(['en', 'ru']);
+    translateService.setDefaultLang('ru');
+    translateService.use('ru');
```
Comment (kirill-malyhin) [inline src/app/language.service.ts:15]: one more improvement, create function init and call it from app.component

File: src/app/converter.service.ts [hunk 12]
```diff
@@ -21,6 +21,11 @@ export class ConverterService {
   getCurrencies(date: string) {
     const subject = new BehaviorSubject(this.currenciesList);
     const url = this.urlService.getUrl(date);
+    this.mapLoadedCurrencies(subject, url);
```
Comment (kirill-malyhin) [inline src/app/converter.service.ts:24]: what kind of subject are your returning? you added unexpected async behavior, just write function, which will return processed object

File: src/app/converter/converter.component.ts [hunk 13]
```diff
@@ -5,7 +5,10 @@ import { ICurrency } from '../models/currency';
 @Component({
   selector: 'app-converter',
   templateUrl: './converter.component.html',
-  styleUrls: ['./converter.component.scss']
+  styleUrls: [
+    './converter.component.scss',
+    '../../assets/styles/scss/common.scss'
```
Comment (kirill-malyhin) [inline src/app/converter/converter.component.ts:10]: wrong way of doing such thinks. common.scss should be added into styles.scss in that case

File: src/app/converter/converter.component.ts [hunk 14]
```diff
@@ -79,24 +82,24 @@ export class ConverterComponent implements OnInit {
...
-    const savedCurrenciesAbbreviations = sessionStorage.getItem('selectedCurs');
+    const savedCurrenciesAbbreviations = sessionStorage.getItem('selectedCurrencies');
     if (savedCurrenciesAbbreviations) {
       this.selectedCurrenciesAbbreviations = JSON.parse(savedCurrenciesAbbreviations);
       this.selectedCurrenciesAbbreviations.forEach(item => {
         const foundCurrencyInCurrenciesList = this.converterService.findCurrency(this.currenciesList, item);
         if (foundCurrencyInCurrenciesList) {
-          const foundSelectedCurrency = this.selectedCurrencies.find(item => item === foundCurrencyInCurrenciesList);
-          if (!foundSelectedCurrency) {
+          const SelectedCurrency = this.selectedCurrencies.find(item => item === foundCurrencyInCurrenciesList);
```
Comment (kirill-malyhin) [inline src/app/converter/converter.component.ts:101]: why first letter is upper?

File: src/app/header/header.component.html [hunk 15]
```diff
@@ -18,12 +18,12 @@
...
                         <button class="btn btn-default dropdown-toggle" type="button" id="dropdownMenu1"
                             data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
-                            <img src="../../assets/styles/pictures/pngegg.png" alt="language">
+                            <img src="../../assets/styles/pictures/languageSelectionIcon.png" alt="language">
                             <span class="caret"></span>
                         </button>
                         <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
-                            <li (click)="changeLang('ru')">Русский</li>
-                            <li (click)="changeLang('en')">English</li>
+                            <li (click)="changeLang(eLanguagesList.russian)">Русский</li>
```
Comment (kirill-malyhin) [inline src/app/header/header.component.html:25]: value of enum should be UPPER CASE

File: src/app/header/header.component.ts [hunk 16]
```diff
@@ -1,6 +1,11 @@
 import { Component } from '@angular/core';
 import { TranslateService } from '@ngx-translate/core';
 
+enum LanguagesList {
+  russian = 'ru',
```
Comment (kirill-malyhin) [inline src/app/header/header.component.ts:5]: RUSSIAN: 'ru'

File: src/styles.scss [hunk 17]
```diff
@@ -1 +1 @@
-/* You can add global styles to this file, and also import other style files */
+/* You can add global styles to this file, and also import other style files */
```
Comment (kirill-malyhin) [inline src/styles.scss:1]: here you should add common.scss

**Review by kirill-malyhin — CHANGES_REQUESTED** [review]

**Review by kirill-malyhin — APPROVED** [review]

**Pull request merged**